sdk/monitor/monitor-opentelemetry-exporter/src/export/statsbeat/statsbeatExporter.ts

```diff
@@ -38,6 +38,24 @@ export class AzureMonitorStatsbeatExporter
     });
   }
 
+  /**
+   * Filter out envelopes with zero metric values to prevent exporting zero counts.
+   * This ensures zero counts are observed for internal cleanup but not exported to Azure Monitor.
+   * @param envelopes - Array of telemetry envelopes to filter
+   * @returns Filtered array of envelopes with non-zero metric values
+   */
+  private filterZeroValueMetrics(envelopes: Envelope[]): Envelope[] {
```

Comment:
Can we have this as a separate PR? Seems like we need this logic regardless of customer stats